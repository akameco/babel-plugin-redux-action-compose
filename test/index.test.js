// @flow
import * as fs from 'fs'
import path from 'path'
import { transformFileSync } from 'babel-core'
import plugin from '../src'

const actionPath = path.resolve(__dirname, 'action')

for (const dir of fs.readdirSync(actionPath)) {
  test(dir, () => {
    const input = path.join(actionPath, dir, 'actions.js')

    const { code } = transformFileSync(input, {
      babelrc: false,
      plugins: [
        [plugin, { inputPath: path.join(actionPath, dir, 'other/actions.js') }],
      ],
    })
    expect(code).toMatchSnapshot(`${dir}`)
  })
}
