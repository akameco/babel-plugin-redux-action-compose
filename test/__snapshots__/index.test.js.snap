// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`already-imported 1`] = `
"// @flow
import type { Action as Other2Action } from './other2/actions';

export type Action = Other2Action;"
`;

exports[`base-action 1`] = `"// @flow"`;

exports[`import-action 1`] = `"// @flow"`;

exports[`real-world 1`] = `
"// @flow
import type { Action as Other2Action } from './other2/actions';
import type { Action as Other3Action } from './other3/actions';

import type { Action as OtherAction } from './other/actions';

export type Action = Other2Action | Other3Action | OtherAction;"
`;

exports[`remove 1`] = `""`;
